src/sentry/integrations/discord/message_builder/base/component/base.py

```diff
@@ -14,6 +20,5 @@ class DiscordMessageComponent:
     def __init__(self, type: int) -> None:
         self.type = type
 
-    def build(self) -> dict[str, object]:
-        attributes = vars(self).items()
-        return {k: v for k, v in attributes if v is not None}
```

Comment:
This pattern of getting all attributes of the current class and turning it into a dictionary - 
```
        attributes = vars(self).items()
        return {k: v for k, v in attributes if v is not None}
```
was very common for all of the Discord components. Unfortunately, it's also obscure enough such that mypy isn't able to infer the type further from` dict[str,Any]`.  As a result, we need to explicitly create and set all the fields for these components to conform to the typeddict and also just improve clarity.